### DISCUSSION TITLE
How do I tell if 20dB transmit power is being used?

### DISCUSSION THREAD

**Discussion by Daniel15** [Koenkk/zigbee2mqtt#12923] [General]
- Zigbee2MQTT version: 1.25.2
- Coordinator: Sonoff Zigbee 3.0 USB Dongle Plus
- Firmware version: Koenkk 20211217

I changed `transmit_power` to 20, and I see 20 in the UI under Settings &rarr; Advanced, but LQI in the devices list and in the map didn't change at all. How do I check that it's actually using 20dB transmit power?

**Comment by Koenkk** [comment]
> but LQI in the devices list and in the map didn't change

That is expected, the LQI indicates the connection strength from device -> coordinator, `transmit_power` influences the coordinator -> device connection strength (which cannot be retrieved)

**Reply by Daniel15** [reply]
@Koenkk Thanks for the quick response!

**Reply by ash286** [reply]
> > but LQI in the devices list and in the map didn't change
> 
> That is expected, the LQI indicates the connection strength from device -> coordinator, `transmit_power` influences the coordinator -> device connection strength (which cannot be retrieved)

I'm afriad I'm not in agreement with @Koenkk. @Daniel15 you can certainly check actual transmit power level in herdsman log (search for 'radioTxPower' keyword in the herdsman log). Follow this article to generate herdsman log: https://www.zigbee2mqtt.io/guide/usage/debug.html#enabling-logging

**Comment by FelisucoVFR** [comment]
@[Daniel15](https://github.com/Daniel15)
Hi, how did you chang transmit_power to 20?
I am unable to find anything

**Reply by Daniel15** [reply]
You need to add it to the `advanced` section in the config YAML file. There's documentation here: https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html

This requires an adapter that supports it (e.g. Sonoff's Zigbee dongle plus which uses a CC2652P chip), plus firmware that supports it (version 20211217 or above)

**Reply by FelisucoVFR** [reply]
Yeap, saw it now.
When I asked I was in ZHA but 2 minutes later moved to Z2M and found the power option.
I have 35 devices in a Sonoff dongle with 2 ZBMINI and 2 CC2531 routers and it is impossible getting them working together.

**Reply by marinus61** [reply]
@Koenkk
I am little on the deep end here.

I have followed your adapter setting and added the serial lines into my configuration.yaml file.

Systems has no issues on check config, however during restart it reports:

2022-08-24 20:35:02.535 ERROR (MainThread) [homeassistant.setup] Setup failed for serial: No setup or config entry setup function defined.

I just want to set dB to 20 since Amazon delivered another unit that they described.
Marinus

**Reply by marinus61** [reply]
@Daniel15

ups I was you I wanted this message to, but thought it was Koenkk, due to the place it was filed :)

I am little on the deep end here.

I have followed your adapter setting and added the serial lines into my configuration.yaml file.

Systems has no issues on check config, however during restart it reports:

2022-08-24 20:35:02.535 ERROR (MainThread) [homeassistant.setup] Setup failed for serial: No setup or config entry setup function defined.
Running 2022 firmware (latest)
I just want to set dB to 20 since Amazon delivered another unit that they described.
Marinus

**Reply by dannystettler** [reply]
> You need to add it to the `advanced` section in the config YAML file. There's documentation here: https://www.zigbee2mqtt.io/guide/configuration/adapter-settings.html
> 
> This requires an adapter that supports it (e.g. Sonoff's Zigbee dongle plus which uses a CC2652P chip), plus firmware that supports it (version 20211217 or above)

@Daniel15 : Do you know if there is a similar setting in the iobroker Zigbee Adapter from Kirov Ilya?
